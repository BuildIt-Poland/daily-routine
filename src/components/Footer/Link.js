import styled from 'styled-components';

import media from '../../styles/media';
import { colorOpaqueBlack, fontWeightBold } from '../../styles/designTokens';
import Hitbox from './Hitbox';

const Link = styled(Hitbox).attrs({
  target: '_blank'
})`
  margin: 0 0.4rem;
  cursor: pointer;
  height: 3rem;
  font-weight: ${fontWeightBold};

  color: ${({ theme }) => theme.secondaryColor};
  background-image: ${({ theme }) => `linear-gradient(180deg, ${theme.secondaryColor} 50%, transparent 0)`};
  background-repeat: repeat-x;
  background-size: 2px 0.2rem;
  background-position: 0 2.2rem;

  ${media.phone`
    color: ${colorOpaqueBlack};
    background-image: linear-gradient(180deg, ${colorOpaqueBlack} 50%, transparent 0);
  `};
`;

export default Link;
