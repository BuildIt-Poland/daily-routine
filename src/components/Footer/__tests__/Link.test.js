import React from 'react';
import { create } from 'react-test-renderer';

import Link from '../Link';

describe('COMPONENT - Footer Link', () => {
  it('renders correctly', () => {
    const component = create(<Link />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
