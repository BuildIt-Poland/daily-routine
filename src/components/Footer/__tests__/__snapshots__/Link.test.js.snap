// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Footer Link renders correctly 1`] = `
.c0 {
  height: 100%;
  min-width: 6rem;
  font-size: 1.4rem;
  color: #222222;
  opacity: 0.4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0 svg {
  display: inline-block;
  margin: 0 0.4rem;
}

<a
  className="c0"
  target="_blank"
/>
`;
